utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,91 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes<const N: usize>(
+        &self,
+        sys_calls: [SysCallName; N],
+    ) -> Option<[usize; N]> {
+        let mut indexes = [0; N];
+        let iter = sys_calls.iter().map(|&sys_call| {
+            self.sys_calls_imports
+                .get(&InvocableSysCall::Loose(sys_call))
+                .map(|&(_, call_indexes_handle)| call_indexes_handle)
+        });
+
+        for (index, maybe_index) in indexes.iter_mut().zip(iter) {
+            match maybe_index {
+                Some(idx) => *index = idx,
                 None => {
-                    log::trace!("Wasm hasn't got either gr_reserve_gas, or gr_reservation_send, or both of them");
-                    return;
+                    log::trace!(
+                        "The following sys-calls must be imported: {missing_sys_calls:?}",
+                        missing_sys_calls = sys_calls.map(|sys_call| sys_call.to_str()),
+                    );
+                    return None;
                 }
             }
-        };
+        }
+
+        Some(indexes)
+    }
 
-        let invocable_sys_call = InvocableSysCall::Precise(SysCallName::ReservationSend);
-        let send_from_reservation_signature = invocable_sys_call.into_signature();
+    /// Generates a function which calls "properly" the given sys-call.
+    fn generate_proper_sys_call_invocation(
+        &mut self,
+        sys_call: SysCallName,
+        func_instructions: Instructions,
+        func_locals: Option<Vec<Local>>,
+    ) {
+        let invocable_sys_call = InvocableSysCall::Precise(sys_call);
+        let signature = invocable_sys_call.into_signature();
 
-        let send_from_reservation_func_ty = send_from_reservation_signature.func_type();
+        let func_ty = signature.func_type();
         let func_signature = builder::signature()
-            .with_params(send_from_reservation_func_ty.params().iter().copied())
-            .with_results(send_from_reservation_func_ty.results().iter().copied())
+            .with_params(func_ty.params().iter().copied())
+            .with_results(func_ty.results().iter().copied())
             .build_sig();
 
+        let func_idx = self.module.with(|module| {
+            let mut module_builder = builder::from_module(module);
+            let idx = module_builder.push_function(
+                builder::function()
+                    .with_signature(func_signature)
+                    .body()
+                    .with_instructions(func_instructions)
+                    .with_locals(func_locals.unwrap_or_default())
+                    .build()
+                    .build(),
+            );
+
+            (module_builder.build(), idx)
+        });
```

Comment:
**Suggestion:** Consider handling the case where `func_locals` is `None` in a more explicit way. Currently, when `func_locals` is `None`, it is replaced with an empty vector. This might lead to unexpected behavior if the function is expected to have local variables but none are provided. Instead, you could return an error or a default set of local variables.
```suggestion
        let func_locals = func_locals.unwrap_or_else(|| {
            log::warn!("No local variables provided for function. Using default set.");
            vec![Local::new(1, ValueType::I32)]
        });

        let func_idx = self.module.with(|module| {
            let mut module_builder = builder::from_module(module);
            let idx = module_builder.push_function(
                builder::function()
                    .with_signature(func_signature)
                    .body()
                    .with_instructions(func_instructions)
                    .with_locals(func_locals)
                    .build()
                    .build(),
            );

            (module_builder.build(), idx)
        });
```

Reply:
useless change